### PR TITLE
C++: support C++11 enum class/struct

### PR DESCRIPTION
**Pull request by Corax26**
The modifications to c.c are directly taken from Geany (commit 6c7f69),
docs/tracking.rst has been updated accordingly.

cxx11enum.cpp unit has also been updated to include enum class/struct's.

### CODE REVIEW

**Comment by b4n** [comment]
One point of 52f09c44690a59aa8b8e6df5e4ebe9df0aae9186 was it was *just* a commit import, meaning it kept date and authorship info :)
I don't care much but as Git allows people to do it properly I'd have kept it, and added a second commit with the test.
You can relatively easily import it by fetching if from my fork (`git fetch https://github.com/b4n/fishman-ctags.git cxx11-classed-enum && git checkout -b b4n/cxx11-classed-enum 52f09c44690a59aa8b8e6df5e4ebe9df0aae9186`).

File: c.c [hunk 1]
```diff
@@ -1927,7 +1933,7 @@ static void processToken (tokenInfo *const token, statementInfo *const st)
 		case KEYWORD_SHORT:     st->declaration = DECL_BASE;            break;
 		case KEYWORD_SIGNED:    st->declaration = DECL_BASE;            break;
 		case KEYWORD_STRING:    st->declaration = DECL_BASE;            break;
-		case KEYWORD_STRUCT:    st->declaration = DECL_STRUCT;          break;
+		case KEYWORD_STRUCT:    checkIsClassEnum (st, DECL_CLASS);      break;
```
Comment (b4n) [inline c.c]: this should be `DECL_STRUCT`, like it was in Geany (https://github.com/b4n/fishman-ctags/commit/52f09c44690a59aa8b8e6df5e4ebe9df0aae9186, and https://github.com/geany/geany/commit/6c7f69578d8e142f5994cc9cf0e0abc83a606a1b)

**Comment by b4n** [comment]
If you want I can make a PR out of https://github.com/b4n/fishman-ctags/tree/cxx11-classed-enum and import any test commit you have (or commit yours on your behalf or mine)

**Comment by Corax26** [comment]
Oh well, this commit breaks KEYWORD_STRUCT so much I'm ashamed of it... Once again, I forgot to run the unit tests before pushing >.<

Anyway, I actually tried to get the commit you mentioned in my previous PR (52f09c4), but I got quite confused (why does it show up in GitHub under fishman's repo, without being accessible from the corresponding Git repo?). So, now that you tell me that it's actually in your repo (which is quite logical actually), problem solved: I just have to add your repo as another remote and cherry-pick the commit. Then I add another commit for the unit test and we're done, is it OK with you?

**Comment by b4n** [comment]
> Anyway, I actually tried to get the commit you mentioned in my previous PR (52f09c4), but I got quite confused (why does it show up in GitHub under fishman's repo, without being accessible from the corresponding Git repo?).

yeah GitHub UI's is pretty confusing when you give it a naked commit ID, it shows it on the repo's origin (or maybe the one on which you post? not sure), not the actual fork.  I have no idea why and it's kinda annoying indeed.

> Then I add another commit for the unit test and we're done, is it OK with you?

Sure, sounds perfect.

**Comment by Corax26** [comment]
Cool, I'll close this PR and open yet another one @_@

**Comment by b4n** [comment]
(if you want you can force-push to an existing PR to replace its content)

**Comment by Corax26** [comment]
Thanks for the tip, I'll do that next time :)

**Pull request closed without merging**